ccc-handlers/src/bonapp.rs

```diff
@@ -63,67 +63,17 @@ enum QueryType {
 	ItemNutrition,
 }
 
-#[instrument]
-async fn parse_response<T>(response: reqwest::Response) -> Result<Json<T>, BonAppProxyError>
-where
-	T: serde::de::DeserializeOwned,
-{
-	let response = response
-		.text()
-		.await
-		.map_err(BonAppProxyError::ResponseAcquisition)?;
-
-	let json = serde_json::from_str(&response);
+use QueryType::*;
 
-	match json {
-		Ok(value) => Ok(Json(value)),
-		Err(e) => Err(BonAppProxyError::ResponseParse(e, response)),
-	}
-}
-
-const fn query_base_url(query_type: &QueryType) -> &str {
-	use QueryType::*;
+#[inline]
+const fn get_query_base_url_and_entity(query_type: &QueryType) -> (&str, &str) {
 	match query_type {
-		Cafe => "https://legacy.cafebonappetit.com/api/2/cafes",
-		Menu => "https://legacy.cafebonappetit.com/api/2/menus",
-		ItemNutrition => "https://legacy.cafebonappetit.com/api/2/items",
+		Cafe => ("https://legacy.cafebonappetit.com/api/2/cafes", "cafe"),
+		Menu => ("https://legacy.cafebonappetit.com/api/2/menus", "cafe"),
```

Comment:
"cafe" or "menu"?

Reply:
The first string in this tuple is the base url, and the second is the QSP that gets populated. I am essentially combining two match statements… one from this function and the other from the former `query_url` function.

When we query for cafes or menus, we use the cafe QSP.